src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -197,4 +238,115 @@ private String getStaticFieldRefName(StaticFieldRef staticFieldRef) {
         return staticFieldRef.getField().getName();
     }
 
+    /*
+     * Returns the name of the JArrayRef
+     */
+    private String getJArrayChain(ValueBox valueBox){
+        String nextKey = ((JArrayRef) valueBox.getValue()).getBase().toString();
+        for (HashMap<String, StaticFieldRef> auxMap : getHashMapStatic()) {
+            for (String mapKey : auxMap.keySet()) {
+                if (mapKey.equals(nextKey)) {
+                    return auxMap.get(mapKey).toString();
+                }
+            }
+        }
+        return "";
+    }
+
+    /*
+     * Returns the generated hashmap chain from a key
+     */
+    private String getJInstanceFieldChain(String nextKey){
+        List<HashMap<String, JInstanceFieldRef>> auxValuesHashMap = new ArrayList<>();
+        auxValuesHashMap.addAll(getHashMapJInstanceField());
+
+        //If nextKey not contain $stack is because simple key
+        if (!(nextKey.contains("$stack"))){
+            return nextKey;
+        }
+
+        JInstanceFieldRef currentField = null;
+
+        //The second position is the field called
+        String currentUniqueKey = "<"+nextKey.split(".<")[1];
+
+        //The first key comes before ".<" if you have a stack as a substring
+        nextKey = nextKey.split(".<")[0];
+        boolean isNextKey = true;
+        while(auxValuesHashMap.size()>0 && isNextKey) {
+            isNextKey = false;
+            for (HashMap<String, JInstanceFieldRef> auxMap : getHashMapJInstanceField()) {
+                for (String mapKey : auxMap.keySet()) {
+                    if (mapKey.equals(nextKey)) {
+                        currentField = auxMap.get(mapKey);
+                        isNextKey = true;
+                        auxValuesHashMap.remove(auxMap);
+                    }
+                }
+                if (isNextKey) {
+                    break;
```

Comment:
@galilasmb couldn't you avoid this by using a while instead of a for?

Reply:
It was just not to continue seeing the elements of for each, if you have already formed the chain.

Reply:
I used iterator and a while now.